### DISCUSSION TITLE
Discrepancy in the budget optimization's solution and objective_function

### DISCUSSION THREAD

**Discussion by xijianlim** [google/lightweight_mmm#37] [General]
Hello, I've tried breaking down the optimization script and I think i've found an issue.

With the examples in place, the KPI post-optimization is determined by scipy's optimize.minimize function and provided in the dictionary value **solution['fun']**; while the kpi_without_optim is derived from the _objective_function. 

However, when I use the solution's x values (as in the optimized media values) in the _objective_function, i get very discrepant values: the predicted value is lower than the kpi_without_optim values and the solution:

![image](https://user-images.githubusercontent.com/25444256/178087560-5b791195-1ee2-44d7-bd74-9e3ff456600e.png)

**Comment by xijianlim** [comment]
Update:

So I discovered what was the issue: I should have used the solution.x variables and not converted it by its price equivalency:

![image](https://user-images.githubusercontent.com/25444256/178181295-6cc8ee37-7c0b-4f85-b3b1-ea917e448fd7.png)

**Reply by pabloduque0** [reply]
Hello @xijianlim! Apologies for the late reply!

Yes, exactly you are right, the model is only fed the media variables without the price. We use the price for making sure the budget constraints are met.
